### PR TITLE
Fix/fwn 1037/flyout lag

### PR DESCRIPTION
**Pull request by tbuchann**
## Description (optional)
Reduce the amount of renders by doing a deep comparison of the props

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/Brokerage/BuySellLimitReached.tsx [hunk 1]
```diff
@@ -15,48 +15,46 @@ const Subcontent = styled(Text)`
...
-    <FlyoutContainer>
-      <FlyoutHeader mode='close' data-e2e='BSLimitReachedHeader' onClick={handleClose} />
-      <FlyoutContent mode='middle'>
-        <div style={{ textAlign: 'center' }}>
-          <Image
-            width='48px'
-            height='48px'
-            name='world-alert'
-            srcset={{ 'world-alert2': '2x', 'world-alert3': '3x' }}
+const BuySellLimitReached: React.FC<Props> = ({ handleClose, limitNumber }) => (
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/components/Flyout/Brokerage/BuySellLimitReached.tsx:18]:
```suggestion
const BuySellLimitReached = ({ handleClose, limitNumber }: BuySellLimitReachedProps) => (
```

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/Brokerage/BuySellLimitReached.tsx [hunk 2]
```diff
@@ -15,48 +15,46 @@ const Subcontent = styled(Text)`
...
+          fullwidth
+        >
+          <FormattedMessage id='buttons.ok' defaultMessage='OK' />
+        </Button>
+      </FlyoutFooter>
+    </FlyoutContent>
+  </FlyoutContainer>
+)
 
 export type Props = {
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/components/Flyout/Brokerage/BuySellLimitReached.tsx:59]:
```suggestion
export type BuySellLimitReachedProps = {
```

File: packages/blockchain-wallet-v4-frontend/src/components/Flyout/index.tsx [hunk 3]
```diff
@@ -156,28 +157,34 @@ export const StickyHeaderFlyoutWrapper = styled(FlyoutWrapper)`
...
-        >
-          <FlyoutChildren>
-            {/* Each child must be wrapped in FlyoutChild for transitioning to work */}
-            {children}
-          </FlyoutChildren>
-        </FlyoutModal>
-      ) : null}
-    </AnimatePresence>
-  )
+class Flyout extends React.Component<Props> {
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/components/Flyout/index.tsx:160]: This should be a functional component, wrapped in `memo` (Like `React.memo`), with this function: `(nextProps) => !equals(this.props, nextProps)` as a second parameter, like:

```tsx
export default memo(Flyout, (nextProps) => !equals(this.props, nextProps))
```

Then you don't need to use a class component

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/template.loading.tsx [hunk 4]
```diff
@@ -5,23 +5,20 @@ import styled from 'styled-components'
...
-  return (
-    <Wrapper>
-      <BlockchainLoader width='80px' height='80px' />
-      <Text weight={600} color='grey600' style={{ marginTop: '24px' }}>
-        <FormattedMessage id='modals.simplebuy.processing' defaultMessage='Processing…' />
-      </Text>
-    </Wrapper>
-  )
-}
+const Loading = React.memo(() => (
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/template.loading.tsx:15]: Import `memo` destructured

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/template.success.tsx [hunk 5]
```diff
@@ -1,30 +1,35 @@
...
 import Unsupported from './template.unsupported'
 
-const Success: React.FC<Props> = (props) => {
-  const isUserEligible = props.pairs.length && props.eligibility.eligible && props.fiatCurrency
-  const isUserSddEligible = props.sddEligible && props.sddEligible.eligible
-  const userHitMaxPendingDeposits =
-    props.eligibility.maxPendingDepositSimpleBuyTrades ===
-    props.eligibility.pendingDepositSimpleBuyTrades
+class Success extends React.Component<Props> {
+  shouldComponentUpdate = (nextProps) => !equals(this.props, nextProps)
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/template.success.tsx:11]: Same thing as above

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/index.tsx [hunk 6]
```diff
@@ -12,16 +13,18 @@ import Loading from '../template.loading'
...
     if (this.props.fiatCurrency && !Remote.Success.is(this.props.data)) {
       this.props.buySellActions.fetchPairs({ currency: this.props.fiatCurrency })
       this.props.buySellActions.fetchFiatEligible(this.props.fiatCurrency)
       this.props.buySellActions.fetchSDDEligibility()
-      this.props.buySellActions.fetchOrders()
+      this.props.buySellActions.fetchBSOrders()
     }
   }
 
+  shouldComponentUpdate = (nextProps) => !equals(this.props, nextProps)
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/index.tsx:26]: Here you can keep this one, because the component is already a class component, but it would be nice if you transformed this one into a FC

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/PriceMovement/index.tsx [hunk 7]
```diff
@@ -41,14 +42,16 @@ const getColorFromMovement = (movement: PriceMovementDirType) => {
...
-class PriceMovement extends PureComponent<Props, State> {
+class PriceMovement extends React.Component<Props> {
   componentDidMount() {
     if (!Remote.Success.is(this.props.data)) {
       const { coin } = this.props
       this.props.miscActions.fetchPriceChange(coin, this.props.fiat || 'EUR', TimeRange.DAY)
     }
   }
 
+  shouldComponentUpdate = (nextProps) => !equals(this.props, nextProps)
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/PriceMovement/index.tsx:53]: Same thing

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/CryptoItem/template.success.tsx [hunk 8]
```diff
@@ -63,55 +65,56 @@ const PlusMinusIconWrapper = styled.div`
...
+                  background={coin}
+                />
+              </IconBackground>
+            </PlusMinusIconWrapper>
+          </>
+        )}
+      </CheckoutDisplayContainer>
+    )
+  },
+  (prevProps, nextProps) => equals(prevProps, nextProps)
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/CryptoItem/template.success.tsx:117]: Yep, you did it here

**Review by pedroapfilho — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/buySell/sagas.ts [hunk 9]
```diff
@@ -667,7 +667,6 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
   const fetchSDDEligible = function* () {
     try {
       yield put(A.fetchSDDEligibleLoading())
-      yield call(waitForUserData)
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/buySell/sagas.ts:670]: why is this removed?

Comment (tbuchann) [reply packages/blockchain-wallet-v4-frontend/src/data/components/buySell/sagas.ts:670]: I think the `isTier2` call invoke the `waitForUserData`

**Review by pedroapfilho — APPROVED** [review]
